### ISSUE TITLE
Binance cache/db behaviour

### ISSUE THREAD

**Issue by a7e8** [rotki/rotki#2474]
When I open Rotki the trades from Binance are missing until the job to retrieve the trades has completed again.

I would expect that Rotki stores these trades and deposits etc from the last time the job ran and they'd be available immediately. If the job runs again it would detect and not overwrite a previous import for the same transaction.

Does Rotki store the binance trades locally?

**Comment by LefterisJP** [comment]
Hey @a7e8 rotki does cache trades locally. But until v1.14.2 we were doing something in a silly way. When you went to the trades page all new trades only were queried. For example if you had not opened rotki for a day, only the range of trades of the last 24 hours are queried. For most exchanges this is super fast.

For Binance it's really really slow. The range makes no difference. You always need to query all possible market pairs (it's a really really stupid limitaton on binance side which is pissing me off and nobody from that exchange wants to fix it: https://github.com/rotki/rotki/issues/1073)

But in the next release we are improving this. Before doing the new query for the next 24 hours in the above example we will query the DB and show all we already got to the user first, and only then do the new query.

Which should at least considerably improve user experience.

Implemented in:

- https://github.com/rotki/rotki/pull/2454
- https://github.com/rotki/rotki/pull/2457

Will be rolled out in the next release.

I think this should handle your issue, so I am closing this. If not then feel free to comment below. Or as always join our discord and chat with us: https://discord.gg/aGCxHG7
